src/resolve/test-resolved-etc-hosts.c

```diff
@@ -62,6 +65,17 @@ static void test_parse_etc_hosts(void) {
         assert_se(memcmp(&bn->addresses[2]->address.in6,
                          &(struct in6_addr) { .s6_addr = {0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 5} }, 16 ) == 0);
 
+
+        assert_se(bn = hashmap_get(hosts.by_name, "dash"));
+        assert_se(bn->n_addresses == 1);
+        assert_se(bn->n_allocated >= 1);
+
+        assert_se(bn->addresses[0]->family == AF_INET);
+        assert_se(memcmp(&bn->addresses[0]->address.in,
+                         &(struct in_addr) { .s_addr = htobe32(0x01020306) }, 4) == 0);
```

Comment:
a slightly sloppy but more easily readable syntax for this could be: `._saddr = inet_addr("1.2.3.6")`...

might be good enough for a test case...